fdbserver/include/fdbserver/IKeyValueStore.h

```diff
@@ -146,6 +147,9 @@ class IKeyValueStore : public IClosable {
 	// of a rollback.
 	virtual Future<Void> init() { return Void(); }
 
+	// Provide a tenant id to tenant name map to support per-tenant data encryption. Currently only used by Redwood.
+	virtual void setTenantPrefixIndex(Reference<TenantPrefixIndex> tenantPrefixIndex) {}
```

Comment:
Is this still needed if the Redwood factory creates the IEncryptionKeyProvider, initializes it, and passes it to Redwood?

Reply:
If this is kept, should the default implementation throw?

Reply:
This is what I want to your feedback. The thing is I need to pass the `tenantPrefixIndex` to storage engine (Redwood), which is maintained by a storage server. However storage engine is not owned by storage server. It is created by worker server, and on storage server reboot, the storage engine instance is pass to the new storage server instance, and the new storage server need to pass its own `tenantPrefixIndex` to the storage engine. So a `setTenantPrefixIndex` API seems to make more sense than having worker server create the `tenantPrefixIndex` and pass to both storage engine and storage server. But if you like the later better I can change to that.

Reply:
Agree. This is another reason we should pass IEncryptionKeyProvider from outside Redwood, so that storage server just need to set the TenantPrefixIndex to the IEncryptionKeyProvider.